### PR TITLE
Advanced GDB Post

### PR DESCRIPTION
**Pull request by tyhoff**

### CODE REVIEW

File: _drafts/advanced-gdb.md [hunk 1]
```diff
@@ -0,0 +1,993 @@
...
+(gdb) bt
+```
+
+## Memory
+
+{:.no_toc}
+
+### Listing Memory Regions
+
+It's possible to list the memory regions of the binary currently being debugging by running the `info files` command.
```
Comment (chrisc11) [inline _drafts/advanced-gdb.md:340]:
```suggestion
It's possible to list the memory regions of the binary currently being debugged by running the `info files` command.
```

File: _drafts/advanced-gdb.md [hunk 2]
```diff
@@ -0,0 +1,993 @@
...
+
+```
+(gdb) x/s 0x2000121c
+0x2000121c <shell_uart_thread+104>:	"shell_uart"
+```
+
+The `find` command can also be useful for finding pointers pointing to arbitrary structs. For example, I want to find all the pointers that contain a reference to the variable `mgmt_thread_data`.
+
+```
+(gdb) find 0x20000000, 0x200117e0, &mgmt_thread_data
```
Comment (chrisc11) [inline _drafts/advanced-gdb.md:437]: good example!

File: _drafts/advanced-gdb.md [hunk 3]
```diff
@@ -0,0 +1,993 @@
...
+
+(gdb) info args
+raw_samples = 0x3128115f
+dft_out = 0x2000a900 <my_stack_area+1344>
+num_samples = 536912536
+```
+
+You can also print all static and global variables in the system by using `info variables`, but that will print _a lot_ of variables out to your screen. It's better to filter through them!
+
+The command `info variables` can optional take a regular expression that will perform a search against the name of the variable.
```
Comment (chrisc11) [inline _drafts/advanced-gdb.md:508]:
```suggestion
The command `info variables` can optionally take a regular expression that will perform a search against the name of the variable.
```

File: _drafts/advanced-gdb.md [hunk 4]
```diff
@@ -0,0 +1,993 @@
...
+(gdb) p num_elements
+$1 = 100
+(gdb) p elements
+$2 = (int *) 0x5575e51f6260
+```
+
+We can print this entire array using one of two ways. First, we can cast it to a `int[100]` array and print that.
+
+```
+(gdb) p (int[100])*elements
```
Comment (chrisc11) [inline _drafts/advanced-gdb.md:736]: interesting, i've always written as `*(int[100] *)elements` but I guess the extra * is not necessary

**Review by chrisc11 — APPROVED** [review]

File: _drafts/advanced-gdb.md (hunk 2 above)
Comment (tyhoff) [reply _drafts/advanced-gdb.md:437]: Thanks! It'd be neat to have something use that to find dangling references to free'd heap allocations.

**Pull request merged**